sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -119,6 +122,7 @@ case class AdaptiveSparkPlanExec(
     if (isFinalPlan) {
       currentPhysicalPlan.execute()
     } else {
+      currentPhysicalPlan = applyPhysicalRules(currentPhysicalPlan, queryStagePreparationRules)
```

Comment:
why is this needed?

Reply:
We have split the pre-planning rules into two groups now: `preprocessingRules` and `queryStagePreparationRules`, the former is necessary before the creation of `AdaptiveSparkPlanExec` because otherwise the initial physical plan will have logical subquery nodes in it. The latter, though, can be deferred to after `AdaptiveSparkPlanExec` creation.

Reply:
Can we do it when assigning the default value of `currentPhysicalPlan`? e.g.
```
@volatile private var currentPhysicalPlan = applyPhysicalRules(initialPlan, queryStagePreparationRules)
```